pkg/controller/metrics/metrics.go

```diff
@@ -540,6 +569,34 @@ func (ca *clusterAccumulator) processCluster(cd *hivev1.ClusterDeployment) {
 	}
 }
 
+func getCloudPlatform(platform hivev1.Platform) string {
```

Comment:
Is it not safe to consult the label `hive.openshift.io/cluster-platform` (`HiveClusterPlatformLabel`) instead?

But if it's not, how about we just move [getClusterPlatform](https://github.com/openshift/hive/blob/374913322b5d94efa5ff0fdcd1a6055df351476f/pkg/controller/clusterdeployment/clusterdeployment_controller.go#L2092-L2111) to a utils package and make it public?